storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/FeatureRowDecoder.java

```diff
@@ -36,60 +41,100 @@ public FeatureRowDecoder(String featureSetRef, FeatureSetSpec spec) {
    * @return boolean
    */
   public FeatureRow decode(FeatureRow encodedFeatureRow) {
-    final List<Field> fieldsWithoutName = encodedFeatureRow.getFieldsList();
+    if (isEncodedV1(encodedFeatureRow)) {
+      // TODO: remove support for v1 feature row in Feast 0.7
+      final List<Field> fieldsWithoutName = encodedFeatureRow.getFieldsList();
+      List<String> featureNames =
+          spec.getFeaturesList().stream()
+              .sorted(Comparator.comparing(FeatureSpec::getName))
+              .map(FeatureSpec::getName)
+              .collect(Collectors.toList());
 
-    List<String> featureNames =
-        spec.getFeaturesList().stream()
-            .sorted(Comparator.comparing(FeatureSpec::getName))
-            .map(FeatureSpec::getName)
-            .collect(Collectors.toList());
-    List<Field> fields =
-        IntStream.range(0, featureNames.size())
-            .mapToObj(
-                featureNameIndex -> {
-                  String featureName = featureNames.get(featureNameIndex);
-                  return fieldsWithoutName
-                      .get(featureNameIndex)
-                      .toBuilder()
-                      .setName(featureName)
-                      .build();
-                })
-            .collect(Collectors.toList());
-    return encodedFeatureRow
-        .toBuilder()
-        .clearFields()
-        .setFeatureSet(featureSetRef)
-        .addAllFields(fields)
-        .build();
+      List<Field> fields =
+          IntStream.range(0, featureNames.size())
+              .mapToObj(
+                  featureNameIndex -> {
+                    String featureName = featureNames.get(featureNameIndex);
+                    return fieldsWithoutName
+                        .get(featureNameIndex)
+                        .toBuilder()
+                        .setName(featureName)
+                        .build();
+                  })
+              .collect(Collectors.toList());
+
+      return encodedFeatureRow
+          .toBuilder()
+          .clearFields()
+          .setFeatureSet(featureSetRef)
+          .addAllFields(fields)
+          .build();
+
+    } else if (isEncodedV2(encodedFeatureRow)) {
+      // Feature Row v2 encoding uses a hashed name as the field name and does not have feature set
+      // reference set.
+      // Decoding reverts the field name to a unhashed string and set feature set reference.
+      Map<String, Value> nameHashValueMap =
+          encodedFeatureRow.getFieldsList().stream()
+              .collect(Collectors.toMap(field -> field.getName(), field -> field.getValue()));
+
+      List<String> featureNames =
+          spec.getFeaturesList().stream().map(FeatureSpec::getName).collect(Collectors.toList());
+
+      List<Field> fields =
+          featureNames.stream()
+              .map(
+                  name -> {
+                    String nameHash =
+                        Hashing.murmur3_32().hashString(name, StandardCharsets.UTF_8).toString();
```

Comment:
What is the length of the hash? Just want to make sure its as small as possible.

Reply:
The code is simply overriding fields here. `encodedFeatureRow` is coverted to builder which means its values are preserved in builder form. Whatever that is not `set` is preserved from `encodedFeatureRow`.

Reply:
8 characters.

Reply:
Assuming that the fields stored in the feature row are float values (32 bit, 4 bytes), this would mean a ~3x increase in space consumption.
@woop @pyalex @khorshuheng

Reply:
Not all fields that are stored are float, there're a lot string as well and int64. So everything is not so bad

Reply:
But I guess that we can safely cut hash string to 4-5 chars